### ISSUE TITLE
ParseReact breaks Parse.User.logOut()

### ISSUE THREAD

**Issue by lrettig** [ParsePlatform/ParseReact#161]
ParseReact (0.5.0) patches `Parse.User.logOut()` (Parse 1.6.14) [like so](https://github.com/ParsePlatform/ParseReact/blob/master/src/ParsePatches.js#L82):

    logOut: function() {
      var promise = oldLogOut();
      LocalSubscriptions.currentUser.update();
      return promise;
    },

Unfortunately, this breaks logging out since `LocalSubscriptions.currentUser.update()` runs before `oldLogOut()` has finished and its promise has resolved. It calls `Parse.User.currentAsync()` -- which returns the still logged-in user -- and then on [line 97](https://github.com/ParsePlatform/ParseReact/blob/master/src/LocalSubscriptions.js#L97) it calls `setCurrentUser(current)` which undoes the log out and logs the user right back in.

This it's impossible to log out the Parse user using `Parse.User.logOut()` with ParseReact installed. Help!

**Comment by thewizster** [comment]
+1

ParseReact is seriously in question now with Parse shutting down. Plus, this repo has many issues and no attention for months now. I would recommend migrating away from ParseReact. React is a great platform but unfortunately Parse integration with React is no longer a viable reality.

I may attempt to patch some of these issues I am experience myself in the interim. If I am successful I will post back here. Otherwise, here are some alternatives I am looking into.

https://www.npmjs.com/package/angular-parse
http://backbonejs.org/#View

**Comment by lrettig** [comment]
Shortly after encountering this issue, which was sort of the final straw for me, I moved away from Parse-React and haven't looked back. It wasn't difficult. Instead of using the Parse-React style local subscriptions, I just created a flux-style Store per data type and have each of my React components subscribe to them. Problem solved.

The other thing that was killing me about Parse-React was that I had two completely different data types floating around in my app: Parse Objects and whatever object type Parse-React subscriptions produced.

**Comment by thewizster** [comment]
Thanks for the feedback! That sounds like a viable solution for me as well. @lrettig

**Comment by tristanbbq** [comment]
@lrettig: same here, it's unfortunate, but I'm moving away from parse-react as well.

@thewizster: with parse-server Parse integration in React is still viable!

**Comment by kkgelu** [comment]
Oops, same here, we are doing React Native. Can repro on some devices, users just can't sign out at all.

Large part of the app already built with ParseReact, some workarounds would be appreciated before I rewrite the parts....

**Comment by lrettig** [comment]
@kkgelu you could try disabling A+ compliance on parse promises. That's an option but it's enabled by default now. Note that I haven't tested this.

**Comment by kkgelu** [comment]
@lrettig yeah, don't brother. I started making my flux stores, I am also having the other issue of Parse objects vs. ParseReact-ed objects, frustrating...

**Comment by juanruizcasas** [comment]
bien
